File: frontend/src/app/shared/models/movie.dto.ts
export interface GenreDto {
  id: number;
  name: string;
}

export interface MovieDto {
  id: number;
  title: string;
  overview: string;
  release_date: string;
  poster_path: string | null;
  backdrop_path: string | null;
  vote_average: number;
  vote_count?: number;
  popularity?: number;
  adult?: boolean;
  original_language?: string;
  original_title?: string;
  video?: boolean;
  runtime?: number;
  genres?: GenreDto[];
}

File: backend/src/main/java/web/example/com/tmdb/components/tmdb/business/entity/MovieDto.java
package web.example.com.tmdb.components.tmdb.business.entity;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.List;

/**
 * Data Transfer Object for Movie information from TMDB API
 */
@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class MovieDto {

    @JsonProperty("id")
    private Long id;

    @JsonProperty("title")
    private String title;

    @JsonProperty("overview")
    private String overview;

    @JsonProperty("release_date")
    private String releaseDate;

    @JsonProperty("poster_path")
    private String posterPath;

    @JsonProperty("backdrop_path")
    private String backdropPath;

    @JsonProperty("vote_average")
    private Double voteAverage;

    @JsonProperty("vote_count")
    private Integer voteCount;

    @JsonProperty("popularity")
    private Double popularity;

    @JsonProperty("adult")
    private Boolean adult;

    @JsonProperty("original_language")
    private String originalLanguage;

    @JsonProperty("original_title")
    private String originalTitle;

    @JsonProperty("video")
    private Boolean video;

    @JsonProperty("runtime")
    private Integer runtime;

    @JsonProperty("genres")
    private List<GenreDto> genres;

    @Data
    @Builder
    @NoArgsConstructor
    @AllArgsConstructor
    public static class GenreDto {
        private Integer id;
        private String name;
    }
}
